code/game/machinery/computer/mechlaunchpad.dm

```diff
@@ -0,0 +1,150 @@
+/obj/machinery/computer/mechpad
+	name = "mecha orbital pad console"
+	desc = "Sends mechs through space to space. Why would you do that?"
+	icon_screen = "mechpad"
+	icon_keyboard = "teleport_key"
+	circuit = /obj/item/circuitboard/computer/mechpad
+	ui_x = 475
+	ui_y = 130
+	///Selected mechpad in the console
+	var/selected_id
+	///Mechpads that it can send mechs through to other mechpads
+	var/obj/machinery/mechpad/connected_mechpad
+	///List of mechpads connected
+	var/list/obj/machinery/mechpad/mechpads = list()
+	///Maximum amount of pads connected at once
+	var/maximum_pads = 3
+
+/obj/machinery/computer/mechpad/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		connected_mechpad = connect_to_pad()
+		connected_mechpad.connected_console = src
+		for(var/obj/machinery/mechpad/pad in world)
+			if(pad == connected_mechpad)
+				continue
+			mechpads |= pad
+			if(LAZYLEN(mechpads) < maximum_pads)
+				break
+
+/obj/machinery/computer/mechpad/Destroy()
+	if(connected_mechpad)
+		connected_mechpad.connected_console = null
+	return ..()
```

Comment:
```suggestion
	if(connected_mechpad)
		connected_mechpad.connected_console = null
		connected_mechpad = null
	return ..()
```
What about `mechpads`? Are you keeping those references after deletion?
What happens if `connected_mechpad` or contents in `mechpads` qdelete before the computer?

Reply:
so what am i supposed to do

Reply:
There's more than one way to do this.
If the elements inside `mechpads` have a reference to the thing that keeps the list, they could remove themselves on Destroy:
```cpp
/obj/machinery/mechpad/Destroy()
	for(var/c in connected_to)
		var/obj/machinery/computer/mechpad/mechpad_computer = c
		mechpad_computer.mechpads -= src
	connected_to = null
	return ..()
```
If they don't, you can register to their deletion on adding to the list and unregister on removing:
```cpp
/obj/machinery/computer/mechpad/mechpad_computer/proc/add_to_mechpads(new_mechpad)
	RegisterSignal(new_mechpad, COMSIG_PARENT_QDELETING, .proc/on_mechpad_qdeletion)
	mechpads += new_mechpad

/obj/machinery/computer/mechpad/mechpad_computer/proc/remove_from_mechpads(old_mechpad)
	UnregisterSignal(old_mechpad, COMSIG_PARENT_QDELETING)
	mechpads += old_mechpad

/obj/machinery/computer/mechpad/mechpad_computer/proc/on_mechpad_qdeletion(obj/machinery/mechpad/source, force)
	remove_from_mechpads(source)
```